docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+- It will next send this EventList into the StorageParser
+- The StorageParser uses its functions to convert the events into string representations
+- The StorageParser then returns these string representations to Storage
+- the Storage class then writes them to the correct Path and the corresponding text(.txt) files are modified. 
+## Implementation
+
+In this section, the features implemented will be explained using various UML diagrams and code snippets.
+
+#### Help feature
+(WIP)
+
+#### Add feature
+
+The add feature in the program allows the user to create one of 3 different events in scheduler--;.
+These 3 are the Personal, Zoom and Timetable events. These events have varying numbers of arguments or fields that can 
+be inserted upon creation. 
+
+Firstly, to begin, the user needs to key in the command `add [event type] [description]; [date]; [time]`
+The optional fields to fill in like the link and location for the zoom and timetable classes can be inserted respectively . For example,
+`add zoom cs2113t meeting; zoom.sg; 16/09/20; 2100`
+
+When a command like this is called, the constructor to `addCommand` will be able to detect the event type based on the user's input. It then stores the event type in that instance of the addCommand.
+
+Next, when `addCommand#execute` is called from the main, this method will call the respective method to create one of the three events. These methods are `addCommand#addPersonal`, `addCommand#addZoom`, and `addCommand#addTimetable`.
+
+The personal event can contain the following fields: 
+1. Description
+2. Description and date
+3. Description, date and time
+
+The zoom event can contain the following fields:
+1. Description and link
+2. Description, link, date and time
+
+The Timetable event can contain the following fields:
+1. Description, date and time
+2. Description, location, date and time
+
+The fields for what each event can contain were chosen based on what we as a team thought were important fields for the respective event types.
+However, these methods can easily be edited to accept different numbers of fields if we change our minds in the future.
+
+Given below is an example scenario of the add feature:
+ 
+Step 1. The user launches the application for the first time. There will be no events stored at the moment.
+ 
+Step 2. The user executes `add zoom cs2113t meeting; zoom.sg; 16/09/20; 2100`. The `addCommand` instance is created and detects that the event is of Zoom type.
+ 
+Step 3. `addCommand#execute` is called. The class knows the current addCommand is of Zoom type so it calls `addCommand#addZoom`.
+ 
+Step 4. `addCommand#addZoom` detects there are 4 fields in the command, separated by semicolons, and uses this to create a new Zoom event.
+ 
+Step 5. The Zoom event is then added to the user's `UserData` for further use.
+ 
+The following sequence diagram shows how the whole add feature works: <br>
+
+![Sequence Diagram for Add Command](./diagrams/addCommand.jpg)
```

Comment:
Consider adding details about what is being returned. The current diagram implies that addCommand, for example, is returning void, which is probably not the case.

Class names should have a capital first letter i.e. AddCommand and not addCommand.

Consider using a general addEvent and general :Event object to represent the 3 classes, as the method calls and invocations are identical except for the names of the object/method. (e.g. diagram drawn in week 10 lecture, where Cat and Dog were represented by a single Animal class)